Userland/Libraries/LibWeb/URL/URL.cpp

```diff
@@ -450,17 +451,18 @@ WebIDL::ExceptionOr<String> URL::hash() const
 }
 
 // https://url.spec.whatwg.org/#ref-for-dom-url-hash%E2%91%A0
-void URL::set_hash(String const& hash)
+WebIDL::ExceptionOr<void> URL::set_hash(String const& hash)
 {
     // 1. If the given value is the empty string:
     if (hash.is_empty()) {
         // 1. Set this’s URL’s fragment to null.
         m_url.set_fragment({});
 
-        // FIXME: 2. Potentially strip trailing spaces from an opaque path with this.
+        // 2. Potentially strip trailing spaces from an opaque path with this.
+        TRY(strip_trailing_spaces_from_an_opaque_path(*this));
```

Comment:
Please add a test for this which would previously fail. Potentially a good spot to add a test would be: https://github.com/SerenityOS/serenity/blob/master/Tests/LibWeb/Text/input/URL/url.html